### DISCUSSION TITLE
Problem installing and launching extensions...

### DISCUSSION THREAD

**Discussion by darth2602** [TheLastBen/fast-stable-diffusion#509] [Q&A]
Hello, today the installation of the extensions does not seem to work properly. 
For Dreambooth for example, the installation seems to work fine but on restart (on google colab) 

I have an error message at the "Start stable-diffusion" step  
(but in web ui, dreambooth is indicated as installed)

Preloading Dreambooth!
Error loading script: main.py
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/scripts.py", line 164, in load_scripts
    module = script_loading.load_module(scriptfile.path)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/script_loading.py", line 13, in load_module
    exec(compiled, module.__dict__)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/main.py", line 3, in <module>
    from dreambooth import conversion, dreambooth
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/conversion.py", line 30, in <module>
    from dreambooth import xattention
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/dreambooth/xattention.py", line 6, in <module>
    from diffusers.models.attention import xformers
ImportError: cannot import name 'xformers' from 'diffusers.models.attention' (/usr/local/lib/python3.7/dist-packages/diffusers/models/attention.py)

**Comment by TheLastBen** [comment]
hi, try `!pip install -U diffusers`

**Reply by darth2602** [reply]
???  i'm on google colab...

**Reply by TheLastBen** [reply]
yes, make a new cell and type ` !pip install -U diffusers` and run it

**Reply by darth2602** [reply]
I fixed the error (thanks Ben !)

i added at the END  "!pip install -U diffusers"    (without "") in cell Installing Requirements ...

BEFORE
_xformers()
cleanup()_

AFTER
_xformers()
cleanup()
!pip install -U diffusers_


But now i have the the message :

_Preloading Dreambooth!
[!] Not using xformers memory efficient attention.
........_



But Dreambooth tab is now on Web UI
the last problem is : xformers()

**Comment by darth2602** [comment]
Dreambooth installation error message :

Error running install.py for extension /content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension.
Command: "/usr/bin/python3" "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/install.py"
Error code: 1
stdout: <empty>
stderr: Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/install.py", line 8, in <module>
    from launch import run
ModuleNotFoundError: No module named 'launch'

**Reply by TheLastBen** [reply]
is your repo updated ?

**Reply by darth2602** [reply]
yes, even totally deleted the SD folder in google colab, it doesn't change anything, finally I deleted Dreambooth from Web Ui which didn't work at all and I test Fast-dreamBooth instead

on the other hand at the stage of the update of the repo I have a very fast message "unable to remove..." I cannot read the continuation, and I have systematically "Up to date"

**Reply by TheLastBen** [reply]
did you check the dreambooth extension's repo ? I read that it is still unstable

**Reply by darth2602** [reply]
yes, after many, may tries, Dreambooth seems to work but crashes after a while, doesn't find my pictures or tells me that the file already exists (?) so I gave up and I try fast dreambooth (not really fast in fact...) it works

**Reply by TheLastBen** [reply]
I called it fast because of xformers, that was the early days of the xformers optimization

**Reply by darth2602** [reply]
I misspoke, it's not fast because I run it on Google Colab and not on a local pc...and colab is not very fast...

**Reply by TheLastBen** [reply]
well the colab pro offers the A100 if you're doing professional stuff

**Reply by darth2602** [reply]
After testing, the extensions that do not install correctly and/or do not work (on colab, not tested on pc) are :

Dreambooth 
Aesthetic Image Scorer
a1111 sd webui tagcomplete
SD Smart process
stable-diffusion-webui-auto-tls-https
Training picker
auto sd paint

**Reply by TheLastBen** [reply]
try them on pc to determine which ones are incompatible with the current colab setup

**Comment by darth2602** [comment]
i cannot , my local pc is very old (10 years)

**Comment by darth2602** [comment]
Error loading script: api.py
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/scripts.py", line 184, in load_scripts
    module = script_loading.load_module(scriptfile.path)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/script_loading.py", line 11, in load_module
    compiled = compile(text, path, 'exec')
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/api.py", line 14
    db_half_model: bool = False,
                               ^
SyntaxError: invalid syntax

LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
Loading weights [a2a802b2] from /content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/model.ckpt
Applying cross attention optimization (Doggettx).
Model loaded.
Loaded a total of 0 textual inversion embeddings.
Embeddings: 
Error executing callback ui_tabs_callback for /content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/main.py
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/script_callbacks.py", line 95, in ui_tabs_callback
    res += c.callback() or []
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd_dreambooth_extension/scripts/main.py", line 201, in on_ui_tabs
    outputs=[db_shuffle_after_epoch]
TypeError: change() missing 1 required positional argument: 'inputs'

**Reply by nawnie** [reply]
some of these do not run on the orignal colab the web ui is from or at least i haven't figured it out. if you want to try there dreambooth just use Shivam as that's what they did and i hear the results are not as good as the original yet. 

i cant really compare these two yet as well sometimes one just works better then the other, and they update so fast and making the models takes a good while. both work well but Shivam will require a little extra setup
